website/blog/2024-04-26-deep-learning-part-2/index.md

```diff
@@ -0,0 +1,287 @@
+conducting training) and optimal model performance rely heavily on preventing a model from overfitting on the training data. A model
+that trains for too long will invariably become quite good at effectively ‘memorizing’ the training data, resulting in a reduction in
+the performance of the model when presented with novel data. A straightforward way to evaluate this behavior is to ensure that
+validation data set metrics (scoring loss metrics on data that is not in the training data set) are captured during the training
+loop. Integrating the MLflow callback into the PyTorch Lightning Trainer allows for iterative logging of loss metrics at
+configurable iterations, enabling an easily debuggable evaluation of the training performance, ensuring that stopping criteria
+can be enforced at the appropriate time to prevent overfitting.
+
+### Evaluating epoch checkpoints of Fine Tuned Models with MLflow
+
+With your training process meticulously tracked and logged by MLflow, you have the flexibility to retrieve and test your model at
+any arbitrary checkpoint. To do this, you can use the mlflow.pytorch.load_model() API to load the model from a specific run
+and use the `predict()` method for evaluation.
+
+In the example below, we will load the model checkpoint from the 3rd epoch and use the `Lightning` train module to generate predictions based on the
+checkpoint state of the saved training epoch.
+
+```python
+import mlflow
+
+
+mlflow.pytorch.autolog(disable = True)
+
+run_id = '<Add the run ID>'
+model = mlflow.pytorch.load_model(f"runs:/{run_id}/model")
+
+model = mlflow.pytorch.load_checkpoint(TransformerModule, run_id, 3)
+
+train_module.predict(model, dataloaders = DataLoader(ds))
+```
+
+## Summary
+
+The integration of MLflow into the finetuning process of pre-trained language models, particularly for applications like custom
+named entity recognition, text classification and instruction-following represents a significant advancement in managing and
+optimizing deep learning workflows. Leveraging the autologging and tracking capabilities of MLflow in these workstreams not only
+enhances the reproducibility and efficiency of model development, but also fosters a collaborative environment where insights
+and improvements can be easily shared and implemented.
+
+As we continue to push the boundaries of what these models can achieve, tools like MLflow will be instrumental in harnessing their full potential.
+
+If you're interested in seeing the full example in its entirety, feel free to [see the full example implementation](https://github.com/puneet-jain159/deeplearning_with_mlfow)
+
+### Check out the code
+
+The code we provide will delve into additional aspects such as training from a checkpoint, integrating MLflow and TensorBoard, and utilizing Pyfunc for model wrapping, among others. These resources are specifically tailored for implementation on [Databricks Community Edition](https://mlflow.org/blog/databricks-ce).
```

Comment:
nit: Can we add link to the repo here? (Ideally a notebook that users can import to DCE)